app/controllers/application_controller/performance.rb

```diff
@@ -702,6 +702,15 @@ def perf_gen_data_before_wait
                              from_dt,
                              to_dt,
                              interval_type]
+      elsif @perf_record.kind_of?(MiddlewareServer)
```

Comment:
Existing logic handles daily/hourly/realtime from different reports.
This part is for daily/hourly types.
Each type of report might have their own layout/metrics.